Add exception stacktrace to logs

vasilievip opened the pull request:
Its hard to guess root cause by having only "Could not fetch" in logs

spencergibb commented:
We've had this discussion before. The multiple git environment will have errors by design (not found). Adding these exceptions will produce false errors and fill logs in some cases. Putting them as debug would let you dynamically change the level to see problems.

spencergibb commented:
I don't agree on changing warns to errors either

vasilievip commented:
Ok, I changed stacktrace logging only when debug is enabled and switched back from error to warn.
Regarding EncryptionController - I found IllegalArgumentException or IllegalStateException cases only on decrypt, don't see where it can be thrown on encrypt - there is key validation done upfront which throws KeyNotInstalledException.

spencergibb commented on spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/JGitEnvironmentRepository.java:
```diff
@@ -484,6 +486,14 @@ private boolean containsBranch(Git git, String label, ListMode listMode)
 
+	protected void warn(String message, Exception ex) {
+		if(logger.isDebugEnabled()) {
+			logger.warn(message, ex);
```
should be `.debug()`.

vasilievip replied:
why? 
if debug enabled new data will be sent but message level will stay as warn. If I search for errors and warnings in kibana I will not find this message anymore, it will be confusing.

spencergibb replied:
It's a standard we follow, we don't check if one level is enabled, then log another. Feel free to log the warn AND then the debug.

vasilievip replied:
done

spencergibb commented:
Closed with some polish via 8c8591cf53755622b7eceee467c638d7e2634baa
